src/stores/userStore.ts

```diff
@@ -0,0 +1,77 @@
+import { create } from 'zustand'
+import { createJSONStorage, persist } from 'zustand/middleware'
+import { User } from '../types/User'
+import { fetchUsers } from '../services/userService'
+
+interface UserState {
+  users: User[]
+  setUsers: (users: User[]) => void
+  initializeUsers: () => Promise<void>
+  updateUser: (updatedUser: User) => Promise<void>
+  deleteUser: (userId: string) => Promise<void>
+}
+
+export const useUserStore = create(
+  persist<UserState>(
+    (set) => ({
+      users: [],
+      setUsers: (users) => set({ users }),
+      initializeUsers: async () => {
+        const dbUsers = await fetchUsers()
+        set({ users: dbUsers })
+      },
+      updateUser: async (updatedUser: User) => {
+        try {
+          // First update the database
+          const response = await fetch(`http://localhost:3300/users/${updatedUser.id}`, {
+            method: 'PUT',
+            headers: {
+              'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(updatedUser)
+          });
+
+          if (!response.ok) {
+            const errorData = await response.text();
+            throw new Error(`Failed to update user: ${errorData}`);
+          }
```

Comment:
i'd say this part has no relation to the user store and would look better as a standalone function